File: test/Codegen.Conformance/classes/preamble-mutable.fs
// A `let mutable` is ONE storage location on the instance, shared by the function-valued
// `let` that writes it and the member that reads it. Two calls are what makes that
// observable: a backend that gave the closure its own copy (a ref cell captured by value,
// a field re-initialised per call) prints 6 twice and never accumulates.
type Counter(step: int) =
    let mutable count = 0
    let bump (k: int) = count <- count + k * step

    member this.Bump(k: int) =
        let _ = bump k
        count

let c = Counter(2)
printfn "%d" (c.Bump 3)
printfn "%d" (c.Bump 4)

File: test/XParsec.FSharp.Codegen.Js.Tests/goldens/preamble-mutable.js
class Counter {
  constructor(step) {
    this.step = step;
    const _s0 = this;
    (this.count = 0);
    (this.bump = (k) => (_s0.count = (((_s0.count) + ((Math.imul((k), (_s0.step))))) | 0)));
  }
}
const Counter__Bump = (_s0) => (k) => (_s0.bump(k), _s0.count);
const c = new Counter(2);
console.log(Counter__Bump(c)(3));
console.log(Counter__Bump(c)(4));
